core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -382,7 +387,35 @@ public CompletableFuture<Void> maybeInitialize() {
                 }
 
                 // Set the state epoch and end offset from the persisted state.
-                startOffset = partitionData.startOffset() != -1 ? partitionData.startOffset() : 0;
+                if (partitionData.startOffset() != PartitionFactory.DEFAULT_START_OFFSET) {
```

Comment:
`partitionData.startOffset()` can't this be -1? If not then how do we differentiate between if the persisted value is zero vs nothing yet persisted?